### PR TITLE
Editor emacs update

### LINKED ISSUE #283: Add emacs to editor customization guide
I wrote some notes for customizing vim, but I didn't have enough time to (adequately) write a comparable emacs introduction. Maybe an emacs user could help us out here? The file in question is in `/static/w17/oh-plus-plus/wk4-ohpp.md`

### PR DESCRIPTION
**Pull request by rarhoade**
# Advanced HW 5 submission

Closes #283

Name: Rahul Rhoades
Uniqname: rarhoade

### CODE REVIEW

**Comment by rarhoade** [comment]
I don't really understand why my pull request failed? The only thing that I edited is the .md file in w17/oh-plus-plus. I saw on piazza that other people have had this issue and I'm pretty sure it's the same error?

**Comment by mterwill** [comment]
Correct; it's not your fault, rather an indication that we forgot to upload materials to the course website. Fixed the issue and re-ran your build. It's passing now.

File: static/w17/oh-plus-plus/wk4-ohpp.md [hunk 1]
```diff
@@ -135,4 +135,70 @@ nmap <leader>erc :e ~/.config/nvim/init.vim<CR>
 * [A laundry list of interesting
 configurations](https://sheerun.net/2014/03/21/how-to-boost-your-vim-productivity/)
 
+## 'emacs'
```
Comment (the-alex) [inline static/w17/oh-plus-plus/wk4-ohpp.md]: Nitpick: These should probably be backticks (\`) like `vim`.

File: static/w17/oh-plus-plus/wk4-ohpp.md [hunk 2]
```diff
@@ -135,4 +135,70 @@ nmap <leader>erc :e ~/.config/nvim/init.vim<CR>
...
+
+# First thing to know
+
+* In order to acces the myriad of commands in Emacs, the notation that begins accessing them begins with either:
+  - 'C -...' which is the same as pressing Ctrl with another key (examples like C -a). There is
+     also the command 'C -x', which prompts for another command to come afterwards (example C -x
+     (C -s).
+  --and--
+  - 'M -...' which is the same as pressin Alt with a key.  There is also 'M -x' and his key is 
+    very important because it prompts the user for command input afterward, in the minibuffer.
```
Comment (the-alex) [inline static/w17/oh-plus-plus/wk4-ohpp.md]: Much like above, I think it'd be better to wrap command-line text or key combinations with backticks. This implies that the command should be typed, and more clearly displays characters. (l and I vs. `l` and `I`)

File: static/w17/oh-plus-plus/wk4-ohpp.md [hunk 3]
```diff
@@ -135,4 +135,70 @@ nmap <leader>erc :e ~/.config/nvim/init.vim<CR>
...
+     Emacs itself because it has built in GUI's for these aspects of the editor.
+
+# First thing to know
+
+* In order to acces the myriad of commands in Emacs, the notation that begins accessing them begins with either:
+  - 'C -...' which is the same as pressing Ctrl with another key (examples like C -a). There is
+     also the command 'C -x', which prompts for another command to come afterwards (example C -x
+     (C -s).
+  --and--
+  - 'M -...' which is the same as pressin Alt with a key.  There is also 'M -x' and his key is
```
Comment (the-alex) [inline static/w17/oh-plus-plus/wk4-ohpp.md]: s/pressin/pressing/

It's probably worth mentioning that the meta key can change depending on platform. For instance, on my mac emacs is set to `esc` by default.

File: static/w17/oh-plus-plus/wk4-ohpp.md [hunk 4]
```diff
@@ -135,4 +135,70 @@ nmap <leader>erc :e ~/.config/nvim/init.vim<CR>
...
+* If you want to install an outside package, there are two routes you can go down. You can either:
+   - insert code in the init file of emacs in the format of: 
+     (require 'package)
+     (add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/"))
+            If you wanted to get to the melpa archives, for example.
+   - or you can use the command 'M -x customize-variable' and then type 'package-archives' in
+     order to edit archives through a GUI
+
+# Customizing colorscheme
+* Emacs already has a list of built in colorschemes you can choose from, simply from going to the GUI menu through the command 'M -x customize-themes', and you can select some of the pre-canned ones listed.
```
Comment (the-alex) [inline static/w17/oh-plus-plus/wk4-ohpp.md]: You've mentioned the GUI menu a few times. For the sake of this guide you should probably write assuming someone is using the command line version, or officially state somewhere towards the top that you'll be assuming for the sake of this guide that the reader is using the GUI enabled version and point to a download link or the default location for the application on Ubuntu.

**Comment by mterwill** [comment]
@rarhoade: please follow up by this Friday or we'll close the PR and won't be able to give you credit for the assignment.

**Comment by the-alex** [comment]
@rarhoade  please follow up by this Friday or we'll close the PR and won't be able to give you credit for the assignment.

**Comment by rarhoade** [comment]
Hello Matt,
   I'm kinda confused on what you want from me in terms of the backticks
for the key commands. I thought backticks were simply just the ' key
surrounding my text, and I initially had that with all of my key commands?
I think I am missing something here but I've been googling backticks for
like a half hour now and I don't really understand what I'm doing wrong or
what you want from me here?

**Comment by rarhoade** [comment]
I just pushed a fork that had all of the suggested changes, and disregard
my last email as I have figured out what a backtick actually was since.

**Comment by the-alex** [comment]
We can't see the changes that you pushed. If done correctly, they should show up in this thread. Are you sure you are pushing to `editor-emacs-update` on `rarhoade/c4cs.github.io`?

**Comment by rarhoade** [comment]
I made a new branch when I was working. I pushed to my rarhoade/githib.io
and made a pull request like last time I did it. I have a shift at the
dining hall until 7 so I will check when I get back and can properly assess
what went wrong but I thought I did everything correctly.

On Feb 23, 2017 3:16 PM, "Alex Chojnacki" <notifications@github.com> wrote:

We can't see the changes that you pushed. If done correctly, they should
show up in this thread. Are you sure you are pushing to editor-emacs-update
on rarhoade/c4cs.github.io?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/c4cs/c4cs.github.io/pull/326#issuecomment-282108039>,
or mute the thread

**Comment by rarhoade** [comment]
I have submitted a new pull-request, and tripled checked that I did it with
the branch that contains the updated file. Hopefully everything in there is
up to spec!

**Pull request closed without merging**